source/RoslynAnalyzers/Testing/Integration/IntegrationTestForwardCancellationTokenToInvocationsAnalyzer.cs

```diff
@@ -0,0 +1,271 @@
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Octopus.RoslynAnalyzers.Testing.Integration
+{
+    /// <summary>
+    /// Specific analyzer for Octopus Integration Test container classes.
+    /// This will show "information" hint when method call can forward CancellationToken, but hasn't.
+    /// For Octopus Integration Test, CancellationToken is defined in the base class.
+    /// The existing dotnet analyzer, CA2016, handle cases where CancellationToken is found on the method context / containing class, but
+    /// not when definition is in parent class. (IE: IntegrationTest)
+    /// This implementation is based (heavily) on CA2016 implementation. (https://github.com/dotnet/roslyn-analyzers/pull/3641)
+    /// </summary>
```

Comment:
Consider limiting the summary to a short one-sentence description of what the analyzer does, then put all the great context in `<remarks>`.

e.g. summary (feel free to improve): `Hint when integration test method calls could take an inherited CancellationToken`